### PR TITLE
feat(update-is-user-active)

### PR DESCRIPTION
**Pull request by eladyashar**
# update user activity status from user management page
* add route for changing activity status by user id for admins.
* change isActiveToggle prop name from setUserActivityStatus to onToggle.
* make the activity status toggle in the users table work.

### CODE REVIEW

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts [hunk 1]
```diff
@@ -200,6 +200,22 @@ const useUsersManagement = ({ page, rowsPerPage, cellNameSort, setPage }: useUse
 
     const handleCloseDialog = () => setUserDialog({ isOpen: false, info: {} })
 
+    const setUserActivityStatus = (isActive: boolean, userId: string) : Promise<any> => {
+        const setUpdateActivityStatusLogger = logger.setup('GraphQL request to the DB');
```
Comment (RoeiRom) [inline client/src/components/App/Content/UsersManagement/useUsersManagement.ts:204]: The flow is updating user status not the grapghql request

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts [hunk 2]
```diff
@@ -200,6 +200,22 @@ const useUsersManagement = ({ page, rowsPerPage, cellNameSort, setPage }: useUse
...
+        setUpdateActivityStatusLogger.info('started is user active updating', Severity.LOW);
+        return axios.post('users/updateIsUserActiveById', {
+            isActive,
+            userId
+        }).then((result) => {
+            if(result.data)
+                fetchUsers();
```
Comment (RoeiRom) [inline client/src/components/App/Content/UsersManagement/useUsersManagement.ts:211]: Why fetching the users again and not updating the fetches users in the client

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts (hunk 1 above)
Comment (eladyashar) [reply client/src/components/App/Content/UsersManagement/useUsersManagement.ts:204]: to keep single source of truth

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts (hunk 2 above)
Comment (eladyashar) [reply client/src/components/App/Content/UsersManagement/useUsersManagement.ts:211]: to keep single source of truth

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts [hunk 3]
```diff
@@ -200,6 +200,22 @@ const useUsersManagement = ({ page, rowsPerPage, cellNameSort, setPage }: useUse
 
     const handleCloseDialog = () => setUserDialog({ isOpen: false, info: {} })
 
+    const setUserActivityStatus = (isActive: boolean, userId: string) : Promise<any> => {
+        const setUpdateActivityStatusLogger = logger.setup('GraphQL request to the DB');
+        setUpdateActivityStatusLogger.info('started is user active updating', Severity.LOW);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/UsersManagement/useUsersManagement.ts:205]: change the message to be more clearly

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts [hunk 4]
```diff
@@ -200,6 +200,22 @@ const useUsersManagement = ({ page, rowsPerPage, cellNameSort, setPage }: useUse
...
+        return axios.post('users/updateIsUserActiveById', {
+            isActive,
+            userId
+        }).then((result) => {
+            if(result.data)
+                fetchUsers();
+                setUpdateActivityStatusLogger.info('updated is user active successfully', Severity.LOW);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/UsersManagement/useUsersManagement.ts:212]: same

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts [hunk 5]
```diff
@@ -200,6 +200,22 @@ const useUsersManagement = ({ page, rowsPerPage, cellNameSort, setPage }: useUse
...
+        }).catch((error) => {
+            alertError('לא הצלחנו לעדכן את הסטטוס שלך');
+            setUpdateActivityStatusLogger.error(`error in updating is user active ${error}`, Severity.HIGH);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/UsersManagement/useUsersManagement.ts:215]: same

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

**Review by ItayBenMoshe — APPROVED** [review]

**Review by RoeiRom — APPROVED** [review]

**Pull request merged**